src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -329,6 +329,16 @@ public Task<bool> DelayUntilHostReady(int timeoutSeconds = ScriptConstants.HostT
         }
 
         internal static async Task<bool> DelayUntilHostReady(WebHostResolver resolver)
+        {
+            // need to ensure the host startup has been initiated
+            var manager = resolver.GetWebScriptHostManager();
+            var tIgnore = manager.EnsureHostStarted(CancellationToken.None);
+
+            // wait for the host to get into a running state
+            return await manager.DelayUntilHostReady(throwOnFailure: false);
+        }
+
+        internal static async Task DelayUntilEnvironmentReady()
         {
             if (DelayRequests)
```

Comment:
This code was previously conflated with the DelayUntilHostReady check. Now it is broken out into its own middleware.